entrypoint.sh

```diff
@@ -41,43 +57,55 @@ chown nginx:nginx /var/cache/nginx
 mkdir -p /var/tmp/nginx
 chown nginx:nginx /var/tmp/nginx
 
+#create vhost directory
+mkdir -p /etc/nginx/vhosts/
+
+# Copy nginx main conf file.
+cp templates/nginx.conf /etc/nginx/nginx.conf
+
 # Process templates
-for t in /templates/*
+upstreamId=0
+for t in "${DOMAINSARRAY[@]}"
 do
-  dest="/etc/nginx/$(basename $t)"
-  echo "Rendering template of $dest"
-  sed -e "s/\${DOMAIN}/${DOMAIN}/g" \
-      -e "s/\${UPSTREAM}/${UPSTREAM}/" \
-      $t > $dest
+  dest="/etc/nginx/vhosts/$(basename "${t}").conf"
+  echo "Rendering template of $t in $dest"
+  sed -e "s/\${DOMAIN}/${t}/g" \
+      -e "s/\${UPSTREAM}/${UPSTREAMARRAY[upstreamId]}/" \
+      /templates/vhost.sample.conf > "$dest"
+  upstreamId=$((upstreamId+1))
 done
 
+
+for d in "${DOMAINSARRAY[@]}"
+do
+
 # Initial certificate request, but skip if cached
-if [ ! -f /etc/letsencrypt/live/${DOMAIN}/fullchain.pem ]; then
-  letsencrypt certonly \
-    --domain ${DOMAIN} \
-    --authenticator standalone \
-    ${SERVER} \
-    --email "${EMAIL}" --agree-tos
-fi
+  if [ ! -f /etc/letsencrypt/live/"${d}"/fullchain.pem ]; then
+    letsencrypt certonly \
+      --domain "${d}" \
+      --standalone \
+      --email "${EMAIL}" --agree-tos
+  fi
 
 # Template a cronjob to reissue the certificate with the webroot authenticator
-cat <<EOF >/etc/periodic/monthly/reissue
-#!/bin/sh
+  cat <<EOF >/etc/periodic/monthly/reissue-"${d}"
+  #!/bin/sh
 
-set -euo pipefail
+  set -euo pipefail
 
-# Certificate reissue
-letsencrypt certonly --renew-by-default \
-  --domain "${DOMAIN}" \
-  --authenticator webroot \
-  --webroot-path /etc/letsencrypt/webrootauth/ ${SERVER} \
-  --email "${EMAIL}" --agree-tos
+  # Certificate reissue
+  letsencrypt certonly --force-renewal \
+    --webroot \
+    -w /etc/letsencrypt/webrootauth/ \
+    --domain "${d}" \
+    --email "${EMAIL}" --agree-tos
```

Comment:
Same here; this should still be a single script that runs `letsencrypt certonly --force-renewal` with many `--domain` arguments.